packages/amplication-server/src/core/build/build.service.ts

```diff
@@ -688,7 +688,9 @@ export class BuildService {
       if (privatePlugin.version !== "latest") {
         pluginsToDownload.push({
           pluginId: privatePlugin.pluginId,
-          pluginVersion: privatePlugin.version,
+          pluginVersion: privatePlugin.version.includes("dev")
+            ? null
+            : privatePlugin.version,
```

Comment:
to make things more clear - it is better to explicitly tell the service to use the dev version - and keep the logic for selecting the branch etc internally for the service